dozer/bot.py

```diff
@@ -52,47 +61,47 @@ def __init__(self, config):
+        elif isinstance(exception, commands.NotOwner):
+            await context.send('{}, {}'.format(context.author.mention, exception.args[0]))
+        elif isinstance(exception, commands.MissingPermissions):
+            permission_names = [name.replace('guild', 'server').replace('_', ' ').title() for name in exception.missing_perms]
+            await context.send('{}, you need {} permissions to run this command!'.format(
+                context.author.mention, utils.pretty_concat(permission_names)))
+        elif isinstance(exception, commands.BotMissingPermissions):
+            permission_names = [name.replace('guild', 'server').replace('_', ' ').title() for name in exception.missing_perms]
+            await context.send('{}, I need {} permissions to run this command!'.format(
+                context.author.mention, utils.pretty_concat(permission_names)))
+        elif isinstance(exception, commands.CommandOnCooldown):
+            await context.send(
+                '{}, That command is on cooldown! Try again in {:.2f}s!'.format(context.author.mention, exception.retry_after))
+        elif isinstance(exception, (commands.CommandNotFound, InvalidContext)):
             pass  # Silent ignore
         else:
-            await ctx.send('```\n%s\n```' % ''.join(traceback.format_exception_only(type(err), err)).strip())
-            if isinstance(ctx.channel, discord.TextChannel):
-                logger.error('Error in command <{0}> ({1.name!r}({1.id}) {2}({2.id}) {3}({3.id}) {4!r})'.format(
-                             ctx.command, ctx.guild, ctx.channel, ctx.author, ctx.message.content))
+            await context.send('```\n%s\n```' % ''.join(traceback.format_exception_only(type(exception), exception)).strip())
+            if isinstance(context.channel, discord.TextChannel):
+                DOZER_LOGGER.error('Error in command <%d> (%d.name!r(%d.id) %d(%d.id) %d(%d.id) %d)',
+                                   context.command, context.guild, context.guild, context.channel, context.channel,
+                                   context.author, context.author, context.message.content)
             else:
-                logger.error('Error in command <{0}> (DM {1}({1.id}) {2!r})'.format(
-                             ctx.command, ctx.channel.recipient, ctx.message.content))
-            logger.error(''.join(traceback.format_exception(type(err), err, err.__traceback__)))
+                DOZER_LOGGER.error('Error in command <%d> (DM %d(%d.id) %d)', context.command, context.channel.recipient,
+                                   context.channel.recipient, context.message.content)
+            DOZER_LOGGER.error(''.join(traceback.format_exception(type(exception), exception, exception.__traceback__)))
 
     @staticmethod
     def format_error(ctx, err, *, word_re=re.compile('[A-Z][a-z]+')):
+        """TODO: Figure out what this does"""
```

Comment:
This turns an exception into a user-friendly (or -friendlier, at least) error message.